### PR TITLE
Diver Management

### PR DESCRIPTION
**Pull request by nspool**
Divers are now return sorted by intials, alphabetical first.
Delete & Clone buttons removed, New button restored.

### CODE REVIEW

**Comment by ccmoloney** [comment]
Users can Edit a diver - so leave that button.

**Comment by nspool** [comment]
@ccmoloney sorry, I meant delete and clone. Fixed description.

File: api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java [hunk 1]
```diff
@@ -21,16 +22,17 @@
...
-    @QueryHints({@QueryHint(name = HINT_CACHEABLE, value = "true")})
-    List<Diver> findByCriteria(@Param("initials")String initials);
-
+    @Query("SELECT d FROM Diver d WHERE d.initials = :initials")
+    @QueryHints({ @QueryHint(name = HINT_CACHEABLE, value = "true") })
+    List<Diver> findByCriteria(@Param("initials") String initials);
 
     @Override
     @RestResource
+    @Query(nativeQuery = true, value = "SELECT * FROM nrmn.diver_ref d ORDER BY (CASE WHEN initials SIMILAR TO '%[a-zA-Z]' THEN 0 ELSE 1 END), d.initials")
```
Comment (jonescc) [inline api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java:35]: Nice solution.

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java:35]: Need to parameterize schema name - see exampels elsewhere by Alex

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java:35]: SiteRepository has an example

Comment (nspool) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java]: @jonescc updated PR to use `{h-schema}`

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java]: Just noticed we're overriding the findAll query - I'm just checking to make sure this query will work with the Pageable  etc.

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java]: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.at-query.native

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java]: Although I don't think we're actually using the paging at the moment

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/DiverRepository.java]: Looks to work OK with examples etc so perhaps just add the count query .

**Comment by nspool** [comment]
@jonescc added count query. Paging isn't being used atm in the UI but testing with curl eg `curl 'http://localhost:8080/api/divers?page=435&size=1`  it looks to be working with the divers still in the correct order.

**Review by jonescc — COMMENTED** [review]
LGTM.  @utas-tajona ?

**Review by jonescc — APPROVED** [review]

**Pull request merged**